src/components/EditGrower.js

```diff
@@ -38,40 +38,67 @@ const EditGrower = (props) => {
   const [growerUpdate, setGrowerUpdate] = useState(null);
   const [loadingGrowerImages, setLoadingGrowerImages] = useState(false);
   const [saveInProgress, setSaveInProgress] = useState(false);
-
+  const [customImageUrl, setCustomImageUrl] = useState(null);
+  const [customImageUrlError, setCustomImageUrlError] = useState({
+    label: '',
+    error: false,
+  });
+  const regex = /\.(gif|jpe?g|tiff?|png|webp|bmp)$/i;
   useEffect(() => {
     async function loadGrowerImages() {
       if (grower?.id) {
         setLoadingGrowerImages(true);
         const selfies = await growerContext.getGrowerSelfies(grower.id);
         setLoadingGrowerImages(false);
-
         setGrowerImages([
-          ...(grower.imageUrl ? [grower.imageUrl] : []),
-          ...(selfies || []).filter((img) => img !== grower.imageUrl),
+          customImageUrl === '' ? null : customImageUrl,
+          ...(grower.image_url ? [grower.image_url] : []),
+          ...(selfies || []).filter((img) => img != grower.image_url),
         ]);
       }
     }
-
+    setCustomImageUrl(null);
     setGrowerUpdate(null);
     loadGrowerImages();
   }, [grower]);
-
+  function isImgURL(url) {
+    if (typeof url === 'object') {
+      return url === null ? false : regex.test(url.image_url);
+    }
+    return regex.test(url);
+  }
   async function handleSave() {
     if (growerUpdate) {
       setSaveInProgress(true);
       // TODO handle errors
       await growerContext.updateGrower({
-        id: grower.id,
+        // id: grower.id,
```

Comment:
You still need to pass the `id` so that `api/growers.updateGrower()` knows where to route the request.
Just remove the `id` before populating the body of the request in `api/growers.updateGrower()`.

Reply:
Got it, 
I just delete Id property before the fetch and it work 
![image](https://user-images.githubusercontent.com/100769556/223252633-69b1ef20-39af-4551-acab-fa5cb75a3a66.png)
The changes which i do in  one user is working but, i have a little problem.
when i click the button save, did the change and keeping in a loop for an uncaugth promise, 
![image](https://user-images.githubusercontent.com/100769556/223254696-564b5665-2d89-404d-bace-fc9c7e53f1ad.png)
It's like a the state SaveInProgress never set in false again, and keeping in true

Reply:
That's because you're deleting the `id` from an object that is used afterwards by the calling function:
```
  const updateGrower = async (payload) => {
    if (payload.organizationId === 'ORGANIZATION_NOT_SET') {
      payload.organizationId = null;
    }
    await api.updateGrower(payload);
    const updatedGrower = await api.getGrower(payload.id);  // <-- This expects id to still be there
```

You shouldn't modify the object that's passed to the function. Instead, create a new object without `id`:
```
      const { id, ...payload } = growerUpdate;
      const growerQuery = `${TREETRACKER_API}/grower_accounts/${id}`;
      return fetch(growerQuery, {
        method: 'PATCH',
        headers: {
          'content-type': 'application/json',
          Authorization: session.token,
        },
        body: JSON.stringify(payload),
      }).then(handleResponse);
```

Reply:
Thanks, i forgot that function. 
Already done the changes

Reply:
Have you pushed your latest changes? I don't see them.

Reply:
I'm sure  I did the push, in 2 hour I'm going to my house an I'll check my computer.

Reply:
> Have you pushed your latest changes? I don't see them.


I already pushed, sorry that night i did the commit but seems i forgot to push to this branch 
![image](https://user-images.githubusercontent.com/100769556/223759894-7d08be05-e294-4f28-8a83-ca62f622480f.png)
![image](https://user-images.githubusercontent.com/100769556/223759915-61d0a016-cf03-48e9-b2d6-9e2bc4c19c32.png)